src/app/select/select-container/select-container.component.ts

```diff
@@ -1,8 +1,107 @@
 })
-export class MzSelectContainerComponent { }
+export class MzSelectContainerComponent implements AfterViewInit, OnDestroy {
+  @ContentChild(MzSelectDirective) mzSelectDirective: MzSelectDirective;
+  @ContentChild(MzValidationComponent) mzValidationComponent: MzValidationComponent;
+  @ContentChild(NgControl) ngControl: NgControl;
+
+  selectValueSubscription: Subscription;
+  statusChangesSubscription: Subscription;
+
+  ngAfterViewInit() {
+    this.initControlSubscription();
+    this.initSelectSubscription();
+  }
+
+  ngOnDestroy() {
+    this.removeControlSubscription();
+    this.removeSelectSubscription();
+  }
+
+  initControlSubscription() {
+    if (this.ngControl) {
+      this.mzSelectDirective.disabled = this.ngControl.control.disabled;
+
+      this.statusChangesSubscription = this.ngControl.control.statusChanges.subscribe((status: string) => {
+        // to handle enabling/disabling formControl
+        const disabled = status === 'DISABLED';
+        if (disabled !== this.mzSelectDirective.disabled) {
+          this.mzSelectDirective.disabled = disabled;
+          this.mzSelectDirective.handleDisabled();
+        }
+      });
+
+      this.selectValueSubscription = this.ngControl.valueChanges.subscribe((value: any) => {
+        const inputValue = this.mzSelectDirective.inputElement.val();
+        const selectedOptionText = this.mzSelectDirective.selectElement
+          .children('option:selected')
+          .text();
+
+        // synchronize input and select when value changes programmatically
+        if (inputValue !== selectedOptionText) {
+          this.mzSelectDirective.inputElement.val(selectedOptionText);
+
+          const dropdownOptions = this.mzSelectDirective.inputElement
+            .siblings('ul.dropdown-content')
+            .children('li');
+
+          dropdownOptions
+            .removeClass('active selected');
+
+          dropdownOptions
+            .filter((index, element: HTMLLIElement) => element.textContent === selectedOptionText)
+            .filter((index, element: HTMLLIElement) => !element.classList.contains('disabled'))
+            .addClass('active');
+
+          if (this.mzValidationComponent) {
+            this.mzValidationComponent.setValidationState();
+          }
+        }
+      });
+    }
+  }
+
+  initSelectSubscription() {
+    if (this.mzSelectDirective) {
+      this.mzSelectDirective.onUpdate.subscribe(() => this.registerOnBlur());
+      this.registerOnBlur();
+    }
+  }
+
+  registerOnBlur() {
+    this.mzSelectDirective.inputElement.on('blur', () => {
+      if (this.ngControl) {
+        this.ngControl.control.markAsTouched();
+      }
+      if (this.mzValidationComponent) {
+        this.mzValidationComponent.setValidationState();
+      }
+    });
+  }
+
+  removeControlSubscription() {
+    if (this.mzSelectDirective) {
+      this.mzSelectDirective.onUpdate.unsubscribe();
+      this.mzSelectDirective.inputElement.off();
```

Comment:
I think it would be better if we remove only the blur event handler. When calling `off` without selector, It will remove all event handlers that were attached with `.on()`.

Reply:
Should only be called when the select element is removed